android/java/org/chromium/chrome/browser/vpn/billing/InAppPurchaseWrapper.java

```diff
@@ -87,15 +86,13 @@ public boolean isMonthlySubscription(String productId) {
                 || productId.equals(RELEASE_MONTHLY_SUBSCRIPTION);
     }
 
-    public boolean isBillingClientReady() {
-        return mBillingClient != null && mBillingClient.isReady();
-    }
-
     public void startBillingServiceConnection(
```

Comment:
I see we call that method from lots of places, on menu button click, on Preference creation and etc, we need to be sure that we called `endConnection` on the previously created `mBillingClient` otherwise we have a leak per a doc.

Reply:
i am making sure of it based on the logs. basically with new api, i am only starting the connection when it's required and ending it when the purpose is served.